### DISCUSSION TITLE
MQTT publish as JSON

### DISCUSSION THREAD

**Discussion by Wilkware** [philippoo66/optolink-splitter#12] [Ideas]
Hi Phil,

danke nochmal für den VS1 Support!
Glaubst Du man könnte die Polling List auch in einem Rutsch als JSON publishen? So das man 3 Topics hätte => cmnd, resp und data!
Vielleicht als optionales Setting :-)

Danke & Ciao
Heiko

**Comment by philippoo66** [comment]
Hai Heiko!

könnte man bestimmt... ;-) wenn es dir nur um den einen Topic (auf der Ebene) geht, kannst du auch einfach in der Poll Liste `"data/MeinDatenpunktname"` statt `"MeinDatenpunktname"` schreiben (müsste zumin gehen, getestet hab ich das noch nicht).

Wenn es dir explizit um JSON geht, müsste man es tatsächlich einbauen. Das auf einen Rutsch ist kein Problem, weil die Daten für eine eventuelle Viessdata csv eh in einem Buffer stehen. Das write_csv kommt ja auch erst, wenn die Liste ein mal 'durch' gelesen ist.

Aktuell bin ich allerdings etwas 'im Rückstand', weil ich grad 2 Tage an meinem eRoller schrauben musste (mittlerweile mein primäres Fortbewegungmittel für 90..95% aller Strecken und Aufgaben) und meine Hündin eine Fehlbedeckung hatte (so heisst das, wenn sie ungewollt geschwängert wurde) und ich alle 2..3 Tage beim Tierarzt vorstellig werden darf...

**Reply by philippoo66** [reply]
Moin Heiko @Wilkware nochmal!
ich werde beizeiten JSON einbauen, als Optionales setting... ich sag dann Beschied.
Grüsse!

**Reply by Wilkware** [reply]
Moin Phil,
perfekt, freu mich schon drauf :)
Danke & Ciao
Heiko

**Reply by philippoo66** [reply]
moin Heiko @Wilkware !

es gibt jetzt einen branch develop_json. Wenn man den 'main topic' ändert in `mqtt_topic = "Vitodens/SENSOR"` und in der settings_ini den Eintrag `mqtt_json_format = True` setzt, erhält man Folgendes: 

![grafik](https://github.com/user-attachments/assets/0c910818-ba20-4fc8-a700-b593a9306745)

der Topic wird dabei immer 'überschrieben', er läuft sozusagen durch. Das ist wohl die Weise, wie es in Tasmota gehändelt wird.

**--> Kannst du bitte sagen (ggf testen), ob das das ist, was du wolltest, oder ansonsten spezifizieren, wie du es dir wünschst!?**

besten Dank & Grüsse!
Phil 

ps. du musst nur mqtt_util.py ersetzen und den Eintrag `mqtt_json_format = True` in der settins_ini.py ergänzen.

**Reply by Wilkware** [reply]
Hi Phil,

super, schau ich mir am WE an und gebe Dir Feedback! 
Bin gespannt :)

Heiko

**Reply by Wilkware** [reply]
Hi Phil,

sorry, jetzt bin ich erst dazu gekommen!

Aber der neue Branch scheint nur VS2 zu können, oder? Ich habe doch den VS1 genutzt!?

Gruß Heiko

**Reply by philippoo66** [reply]
oh, das hatte ich nicht mehr im Kopf. 

hattest du das denn so gedacht, dass alle Infos nacheinander auf einem Topic kommen? so wie oben gezeigt?

ich glaube die mqtt_util.py sollte auch in deinem Branch funktionieren. nur dann in der settigs_ini noch `mqtt_json_format = True` ergänzen

**Reply by Wilkware** [reply]
Okay, bin wieder auf vs1test zurück und habe die mqtt_util.py drübergebügelt. Geht erstmal, aber trotzdem kommen die Daten einzeln :(

![image](https://github.com/user-attachments/assets/882662c4-9835-4f5e-8870-3e5146d0ba9b)

Meine Erwartung wäre sowas ...

`{"getTempA": 4.000000, "getTempAtp": 4.100000, "getTempAged": 4.200000, "getTempWWist": 42.900002, "getTempWWsoll": 45.000000, "getTempKist": 63.000000, "getTempKtp": 62.700001, "getTempKsoll": 65.000000, "getBetriebszeitStandby": 1193046.500000, "getBrennerStatus": "NOT OK", "getBrennerStatus2": "0", "getBrennerStufe": 1, "getBrennerStarts": 25978, "getBrennerStunden1": 21228.535156, "getBrennerStunden2": 5.455555, "getTempStp": 42.900002, "getTempRaumNorSollM1": 22.000000, "getTempRaumRedSollM1": 3.000000, "getPumpeStatusM1": 0, "getPumpeStatusSp": 1, "getPumpeStatusZirku": 1, "getBetriebArt": "H+WW FS", "getStatusFrostM1": 1, "getSystemTime": 1731402740, "getDaempfung": 128.000000, "getStatusStoerung": 0}`

.. also alle Daten als JSON pro Rückmeldung.

Gruß Heiko

**Reply by philippoo66** [reply]
oki, das hatte ich schon gadacht. ja, mit der aktuellen json Version kommen die Daten einzeln, alle nacheinander auf einen Topic. also brauchen wir unterschiedliche json 'Konfigurationen'. bekommen wir aber auch hin. 

danke für's Testen und die Rückmeldung & Grüsse!
